src/macros.rs

```diff
@@ -0,0 +1,72 @@
+//! Warning: extreme macros
```

Comment:
Also, perhaps we shouldn't dump our macros in a file called `macros.rs`? It'd be better to logically put them in a place that makes sense, in this case for the lua conversion.